finagle-base-http/src/test/scala/com/twitter/finagle/http/MapParamMapTest.scala

```diff
@@ -22,7 +22,7 @@ class MapParamMapTest extends FunSuite {
   }
 
   test("+") {
-    val paramMap = MapParamMap() + ("a" -> "1") + ("b" -> "2") + ("a" -> "3")
+    val paramMap = MapParamMap().setParam("a" -> "1").setParam("b" -> "2").setParam("a" -> "3")
```

Comment:
This is really unfortunate, but `+` is final and returns `Map`.

Reply:
HashMap gets around it somehow.  I think it's with StrictOptimizedMapOps, but I'm not sure.

Reply:
I got mixed up with `+=` on `mutable.Map` -- `+` is not final at all, specifically to allow overriding with a narrower return type.

Reply:
oh, phew!

Reply:
phew indeed